test/listen.test.js

```diff
@@ -156,6 +156,20 @@ test('double listen errors callback with (err, address)', t => {
   })
 })
 
+test('nonlocalhost double listen errors callback with (err, address)', t => {
+  t.plan(4)
+  const fastify = Fastify()
+  t.teardown(fastify.close.bind(fastify))
+  fastify.listen({ host: '::1', port: 0 }, (err, address) => {
+    t.equal(address, `http://${'[::1]'}:${fastify.server.address().port}`)
+    t.error(err)
+    fastify.listen({ host: '::1', port: fastify.server.address().port }, (err2, address2) => {
+      t.equal(address2, null)
+      t.ok(err2)
```

Comment:
t.ok(err2) is basically just a truthy check. 

better do:

```suggestion
      t.equal(err2.code, 'FST_ERR_REOPENED_SERVER')
```